### PR TITLE
Extract: Add npgettext to default keywords map

### PR DESCRIPTION
**Pull request by sudheesh001**
Add the npgettext as a default keyword in the keywords map.
Create test_npgettext in test_extract showing the passing tests.

Fixes https://github.com/python-babel/babel/issues/281
Fixes https://github.com/python-babel/babel/issues/328

### CODE REVIEW

**Comment by sudheesh001** [comment]
@akx I hope this is the fix that's expected.
I closed https://github.com/python-babel/babel/pull/337 and opened this pull request.

**Comment by akx** [comment]
@sudheesh001 Thanks! I'll take a closer look later tonight.

File: babel/messages/extract.py [hunk 1]
```diff
@@ -38,7 +38,8 @@
     'dgettext': (2,),
     'dngettext': (2, 3),
     'N_': None,
-    'pgettext': ((1, 'c'), 2)
+    'pgettext': ((1, 'c'), 2),
+    'npgettext': ((1, 'c', 2, 3))
```
Comment (akx) [inline babel/messages/extract.py]: I think the tuple should be `((1, 'c'), 2, 3)`?

I.e. that the first argument is `c`ontext and 2 and 3 are messages.

File: tests/messages/test_extract.py [hunk 2]
```diff
@@ -60,6 +60,15 @@ def test_nested_comments(self):
                                                ['TRANSLATORS:'], {}))
         self.assertEqual([(1, 'ngettext', (u'pylon', u'pylons', None), [])],
                          messages)
+        buf = BytesIO(b"""\
```
Comment (akx) [inline tests/messages/test_extract.py]: These tests are nice enough, but can you move them to another function, `def test_npgettext(self):` for instance? :)

File: tests/messages/test_extract.py [hunk 3]
```diff
@@ -131,6 +140,17 @@ def test_multiline(self):
         self.assertEqual([(1, 'ngettext', (u'pylon', u'pylons', None), []),
                           (3, 'ngettext', (u'elvis', u'elvises', None), [])],
                          messages)
+        buf = BytesIO(b"""\
```
Comment (akx) [inline tests/messages/test_extract.py]: These tests are nice enough, but can you move them to another function, `def test_npgettext(self):` for instance? :)

**Comment by sils1297** [comment]
@sudheesh001 also the same commit message guidelines apply here too (although they're not that strictly enforced)

**Comment by sudheesh001** [comment]
@akx @sils1297 I hope i've fixed it, i've rebased and updated this PR.

**Comment by akx** [comment]
This looks good!

Thank you for the patch, @sudheesh001 !

**Pull request merged**